### ISSUE TITLE
The app no longer works when Sitefinity is set to use OpenID as the authentication protocol

### ISSUE THREAD

**Issue by kappa7194** [Sitefinity/sitefinity-admin-app-extensions#490]
We updated the app to the latest 14.x version ([tag 14.3.8024.0](<https://github.com/Sitefinity/sitefinity-admin-app-extensions/releases/tag/14.3.8024.0>)) and when started the app is no longer working, the user is redirected to the SWT protocol authentication URL even though Sitefinity is configured to use OpenID:

![image](https://user-images.githubusercontent.com/1249988/221839703-117fdbb1-4a72-44b8-8011-f0ff6df9ade3.png)

The call to https://localhost:44376/sitefinity/authentication/info correctly returns `OpenID` but the Angular app redirects the user to the `/sitefinity/authenticate/swt` endpoint.

We **can not** change the authentication protocol to `Default` (nor to `SimpleWebToken`), we need OpenID.

`git bisect` says that the problem has been introduced with the [pull request 472](<https://github.com/Sitefinity/sitefinity-admin-app-extensions/pull/472>).

[Tag 14.2.7928.0](<https://github.com/Sitefinity/sitefinity-admin-app-extensions/releases/tag/14.2.7928.0>) seems to be the last good version.

Here is the log of `git bisect`:

```
$ git bisect log
git bisect start
# status: waiting for both good and bad commits
# good: [8666c80797592b2e5b6da11841d997a39c8cd578] Update sdk to 14.1.7800
git bisect good 8666c80797592b2e5b6da11841d997a39c8cd578
# status: waiting for bad commit, 1 good commit known
# bad: [c6aeda923fe8489c12575287a10123f617f4d183] Upgrade adminapp sdk to 8024 (#489)
git bisect bad c6aeda923fe8489c12575287a10123f617f4d183
# good: [6beb3491ae0022e12821b07740e84ab4cb541635] Merge pull request #330 from Sitefinity/koserkov/bumpSdk7627
git bisect good 6beb3491ae0022e12821b07740e84ab4cb541635
# good: [6beb3491ae0022e12821b07740e84ab4cb541635] Merge pull request #330 from Sitefinity/koserkov/bumpSdk7627
git bisect good 6beb3491ae0022e12821b07740e84ab4cb541635
# good: [09151614f0c998ec9f8998f36c056bb9b6d9f6a2] Decho/merge deveop patches (#415)
git bisect good 09151614f0c998ec9f8998f36c056bb9b6d9f6a2
# good: [e7faea26fd26e87d92855985714a261403b8a0b7] Update sdk to version 14.2.7922 (#456)
git bisect good e7faea26fd26e87d92855985714a261403b8a0b7
# good: [6ad9336bc3f5f8ac7a874ad29e379f0649a0be65] Update sdk to 14.2.7927 (#469)
git bisect good 6ad9336bc3f5f8ac7a874ad29e379f0649a0be65
# bad: [e5a9f5adb985172ed7df3a1ef09783f33da4e387] Merge master -> patches (#476)
git bisect bad e5a9f5adb985172ed7df3a1ef09783f33da4e387
# bad: [4d050758ae9f18cb2ff8fdb57cd0db29e9aac2e7] Decho merge master to fixes (#472)
git bisect bad 4d050758ae9f18cb2ff8fdb57cd0db29e9aac2e7
# good: [63e173900e5ab60ee7610ab5bc315c649ccf263f] Update sdk to 14.2.7927 (#471)
git bisect good 63e173900e5ab60ee7610ab5bc315c649ccf263f
# first bad commit: [4d050758ae9f18cb2ff8fdb57cd0db29e9aac2e7] Decho merge master to fixes (#472)
```

**Comment by stefankoserkov** [comment]
Hello @kappa7194 

Thank you for contacting us.

Unfortunately, this is a bug in the AdminApp, not in the extension per se. 

The reason why you are not seeing this problem manifest when using Sitefinity is that the AdminApp integrated into Sitefinity, authenticates based on the cookie that is already present, but when running the AdminApp's extensions in the so called stand-alone mode, it cannot use the cookie to authenticate because the cookie is http-only. 

You can workaround this issue for the time being by switching to the "Default" authentication, when developing extensions, and when you build them and deploy them in Sitefinity, they will work with OpenId since the AdminApp's mode will be integrated. I would strongly urge you to switch to the default authentication anyway, since in Sitefinity 14.4 all other authentication mechanisms apart from the default authentication, will be dropped.

I must note here that this bug fix will not be shipped with 14.3.8025.0, it will probably be shipped with 14.3.8026.0 at the earliest. 

The bug has been logged with Id 495242. 

Please let me know if you have any further questions, or please feel free to contact our support team.

Best regards,
Stefan

**Comment by Novkirishki** [comment]
Hello @kappa7194,

As Stefan mentioned the "Default" authentication mechnism is the way to go for Sitefinity and we have dropped it's alternatives in version 14.4. Therefore we will not be fixing the bug mentioned (495242). 

Do not hesitate to contact the Sitefinity support team if you have any further questions. 

I am closing the issue now.

Best,
Niki
